pkgs/build-support/emacs/wrapper.nix

```diff
@@ -228,7 +232,8 @@ runCommand
         --subst-var-by bash ${emacs.stdenv.shell} \
         --subst-var-by wrapperSiteLisp "$deps/share/emacs/site-lisp" \
         --subst-var-by wrapperSiteLispNative "$deps/share/emacs/native-lisp:" \
-        --subst-var-by autoloadExpression "-l cl-loaddefs -l nix-generated-autoload" \
+        --subst-var-by autoloadExpression \
+          "${if withAutoloads then "'-l cl-loaddefs -l nix-generated-autoload'" else "''"}" \
```

Comment:
`lib.optionalString` would be more interesting

Reply:
Updated, but I could not test it because I dont have a Darwin machine.